tsdb/series_file_test.go

```diff
@@ -50,20 +50,33 @@ func TestParseSeriesKeyInto(t *testing.T) {
 	}
 }
 
-func TestParseSeriesKeyMeasurement(t *testing.T) {
+func TestParseSeriesKey(t *testing.T) {
 	tests := []struct {
 		name        string
-		measurement []byte
-		expected    []byte
+		seriesKey   []byte
+		expectedKey []byte
 	}{
-		{name: "invalid measurement", measurement: tsdb.AppendSeriesKey(nil, []byte{}, nil), expected: nil},
-		{name: "valid measurement", measurement: tsdb.AppendSeriesKey(nil, []byte("cpu"), nil), expected: []byte("cpu")},
+		{
+			name:        "invalid zero length series key",
+			seriesKey:   tsdb.AppendSeriesKey(nil, []byte{}, nil),
+			expectedKey: nil,
+		},
+		{
+			name:        "valid series key with tags",
+			seriesKey:   tsdb.AppendSeriesKey(nil, []byte("foo"), models.NewTags(map[string]string{"tag1": "foo"})),
+			expectedKey: []byte("foo"),
+		},
+		{
+			name:        "valid series key with no tags",
+			seriesKey:   tsdb.AppendSeriesKey(nil, []byte("foo"), nil),
+			expectedKey: []byte("foo"),
+		},
 	}
```

Comment:
I'd like two more tests here, with a zero-length `models.NewTags` created
- With an empty map
- With a `nil` map

We've had lots of bugs caused by different handling between `nil` and empty collections.

Reply:
Makes sense, I've expanded these now 👍